### PR TITLE
HW-8: Objects and arrays

### PR DESCRIPTION
**Pull request by lenanova**

### CODE REVIEW

**Comment by lenanova** [comment]
Hi @anatoly-karpovich @VladimirLQA Please help to review this PR.

File: hw8/task2.js [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    const newEnterprise = {
+        id: getNewId(),
+        name,
+        departments: []
+    }
+    enterprises.push(newEnterprise)
+}
+
+function getNewId(id) {
+    id = 0;
```
Comment (anatoly-karpovich) [inline hw8/task2.js]:
```suggestion
function getNewId() {
    let id = 0;
```
You don't need this argument

File: hw8/task2.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+/*
+7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+
+Пример:
+deleteEnterprise(1)
+*/
+
+function deleteEnterprise(companyId) {
+    const companyToDelete = getCompanyById(companyId);
+    enterprises.splice(companyToDelete, 1);
```
Comment (anatoly-karpovich) [inline hw8/task2.js]: splice needs an index, so need to search for it using findIndex method

File: hw8/task2.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    
+    if (!dept1 || !dept2) {
+        throw new Error("Один из отделов не существует!");
+    }
+    
+    if (dept1.id === dept2.id) {
+        throw new Error("Выберите два разных отдела!");
+    }
+
+    const employeesToMove = dept1.employees_count;
```
Comment (anatoly-karpovich) [inline hw8/task2.js:316]: You've forgotten to check if this departments are in one enterprise. Just use getEnterprisesByDepartment method for both departments and compare the enterprise id

**Review by anatoly-karpovich — APPROVED** [review]
Awesome implementation! Looks gorgeous!

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**